packages/relic/lib/src/message/message.dart

```diff
@@ -0,0 +1,53 @@
+import 'dart:async';
+import 'dart:convert';
+
+import '../body/body.dart';
+import '../headers/headers.dart';
+
+abstract class Message {
+  /// The HTTP headers associated with this message.
+  final Headers headers;
+
+  /// Extra context for middleware and handlers.
+  final Map<String, Object> context;
+
+  /// The streaming body of the message.
+  Body body;
+
+  Message({
+    required this.body,
+    required this.headers,
+    this.context = const {},
+  });
+
+  /// Returns the MIME type from the Content-Type header, if available.
+  String? get mimeType {
+    if (body.contentType == null) return null;
+    return "${body.contentType!.mimeType}";
+  }
+
+  /// Returns the encoding specified in the Content-Type header, or null if not specified.
+  Encoding? get encoding {
+    if (body.contentType == null) return null;
+    return body.contentType!.encoding;
+  }
+
+  /// Reads the body as a stream of bytes. Can only be called once.
+  Stream<List<int>> read() => body.read();
```

Comment:
Suggestion: Shouldn't this be a stream of `Uint8List`?

Please double check all places where we return a list of int so that they shouldn't be a `Uint8List`.